src/extension/kernel.ts

```diff
@@ -1119,4 +1119,79 @@ export class Kernel implements Disposable {
   public getPlainCache(cacheId: string): Promise<Uint8Array> | undefined {
     return this.serializer?.getPlainCache(cacheId)
   }
+
+  async runProgram(program?: RunProgramOptions | string) {
+    let programOptions: RunProgramOptions
+
+    if (!this.runner) {
+      return Promise.reject(new Error('No runner available'))
+    }
+
+    if (typeof program === 'object') {
+      programOptions = program
+    } else if (typeof program === 'string') {
+      programOptions = {
+        programName: '',
```

Comment:
I don't think this can be empty. We should probably use `bash` (if we can rely on $PATH) or `/bin/bash` and perhaps check the default shell `$SHELL` instead of `$0` (what's shell is currently running)?